tests/Integration/Behaviour/Features/Scenario/Carrier/carrier_management.feature

```diff
@@ -175,26 +225,219 @@ Feature: Carrier management
       | active       | true                               |
       | delay[en-US] | Shipping delay new                 |
 
+Scenario: Partially editing carrier with handling shipping
+    When I create carrier "carrier1" with specified properties:
+      | name             | Carrier 1                          |
+      | grade            | 1                                  |
+      | trackingUrl      | http://example.com/track.php?num=@ |
+      | position         | 2                                  |
+      | active           | true                               |
+      | delay[en-US]     | Shipping delay                     |
+      | shippingHandling | false                              |
+      | isFree           | false                              |
+      | shippingMethod   | 1                                  |
+      | taxRuleGroup     | US-AL Rate (4%)                    |
+      | rangeBehavior    | 1                                  |
+    When I edit carrier "carrier1" called "newCarrier1" with specified properties:
+      | shippingHandling | true                               |
+    Then carrier "carrier1" should have the following properties:
+      | name             | Carrier 1                          |
+      | shippingHandling | false                              |
+    Then carrier "newCarrier1" should have the following properties:
+      | name             | Carrier 1                          |
+      | shippingHandling | true                               |
+
+Scenario: Partially editing carrier with free shipping
+    When I create carrier "carrier1" with specified properties:
+      | name             | Carrier 1                          |
+      | grade            | 1                                  |
+      | trackingUrl      | http://example.com/track.php?num=@ |
+      | position         | 2                                  |
+      | active           | true                               |
+      | delay[en-US]     | Shipping delay                     |
+      | shippingHandling | false                              |
+      | isFree           | false                              |
+      | shippingMethod   | 1                                  |
+      | taxRuleGroup     | US-AL Rate (4%)                    |
+      | rangeBehavior    | 1                                  |
+    When I edit carrier "carrier1" called "newCarrier1" with specified properties:
+      | isFree | true                               |
+    Then carrier "carrier1" should have the following properties:
+      | name   | Carrier 1                          |
+      | isFree | false                              |
+    Then carrier "newCarrier1" should have the following properties:
+      | name   | Carrier 1                          |
+      | isFree | true                               |
```

Comment:
You tested two use cases when both free shipping and shippingHandling are false initially, which are the to simplest use cases You are missing three alternative scenarios:
- `freeShipping` initially true, `shippingHandling` only is set to true -> `isFree` is automatically turned to `false`
- `shippingHandling` initially true, `freeShipping` only is set to true -> `shippingHandling` is automatically turned to `false`
- when both `shippingHandling` and `isFree` are turned to true an exception is thrown, you need to asert the proper exception with the proper code is thrown

Reply:
Thoses tests are already in the PR:
- `Line freeShipping initially true, shippingHandling only is set to true -> isFree is automatically turned to false` -> Start line 372
- `shippingHandling initially true, freeShipping only is set to true -> shippingHandling is automatically turned to false` -> Start line 391
- `when both shippingHandling and isFree are turned to true an exception is thrown, you need to asert the proper exception with the proper code is thrown` -> Start line 410

Reply:
My bad I didn't read all correctly, perfect!